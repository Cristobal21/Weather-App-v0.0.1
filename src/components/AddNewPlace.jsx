import { useState } from "react"
import { Form } from 'react-bootstrap'
import PropTypes from 'prop-types'


export const AddNewPlace = ({ onNewPlace }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) return
        onNewPlace(inputValue.trim())
        setInputValue('')
    }
    
    return (

        <Form onSubmit={ onSubmit } aria-label="form" >
            <Form.Group>
                <Form.Control 
                    type="text" 
                    placeholder="Busca un lugar, paÃ­s o ciudad"
                    value={ inputValue }
                    onChange={ onInputChange }
                />
            </Form.Group>
        </Form>

    )
}

AddNewPlace.propTypes = {
    onNewPlace: PropTypes.func.isRequired,
}
