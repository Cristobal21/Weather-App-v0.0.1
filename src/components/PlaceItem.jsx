import { ListGroup } from "react-bootstrap"



export const PlaceItem = ({ name }) => {

    const handleClick = () => {
        //Hacer el redireccion a weathergrid o weatherdetail
        console.log('Hola hiciste click en > ', name);
    }

    return (
        <>
            <ListGroup>
                <ListGroup.Item action variant="info" className="mb-2" onClick={ handleClick }>
                    <p>{ name }</p>
                </ListGroup.Item>
            </ListGroup>
        </>
    )
}
