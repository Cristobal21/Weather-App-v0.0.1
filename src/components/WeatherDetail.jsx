import { Container, Row } from "react-bootstrap"
import { useFetchWeather } from "../hooks/useFetchWeather"

export const WeatherDetail = ({ lugar }) => {

    const { clima } = useFetchWeather( lugar )

    const dataWeather = () => {
        
    }


    return (
        <>
            <p>
                {
                    clima.map( data => data.name )
                }
            </p>
        </>
    )
}
