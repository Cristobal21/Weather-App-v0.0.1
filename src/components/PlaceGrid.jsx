import { useFetchPlaces } from "../hooks/useFetchPlaces"
import { PlaceItem } from "./PlaceItem"
import PropTypes from 'prop-types'


export const PlaceGrid = ({ lugar }) => {

    const { lugares } = useFetchPlaces( lugar )
    
    return (
        <>
            <p className="mt-3">{`Última búsqueda realizada: ${ lugar }`}</p>
            <hr />

            <div>
                {
                    lugares.map( (data) => (
                        <PlaceItem
                            key={ data.id }
                            { ...data }
                        />
                    ) )
                }
            </div>
        </>
    )
}


PlaceGrid.propTypes = {
    lugar: PropTypes.string.isRequired
}