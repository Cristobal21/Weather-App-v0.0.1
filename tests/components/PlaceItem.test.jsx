import { render, screen } from "@testing-library/react"
import { PlaceItem } from "../../src/components/PlaceItem"

describe('Pruebas en <PlaceItem/>', () => {

    const name = 'Buin, Santiago Metropolitan, Chile'
    
    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<PlaceItem name={ name }/>)
        expect( container ).toMatchSnapshot()
    })

    test('Debe mostrar el nombre en el componente', () => {
        render(<PlaceItem name={ name }/>)
        expect( screen.getByText( name ) ).toBeTruthy()
    })
})