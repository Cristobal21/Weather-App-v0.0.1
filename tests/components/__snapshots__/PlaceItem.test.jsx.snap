// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PlaceItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="list-group"
  >
    <button
      class="mb-2 list-group-item list-group-item-info list-group-item-action"
    >
      <p>
        Buin, Santiago Metropolitan, Chile
      </p>
    </button>
  </div>
</div>
`;
